rust/operator/src/lib.rs

```diff
@@ -512,6 +527,9 @@ impl NifiState {
             container_builder.add_container_port(METRICS_PORT_NAME, port.parse()?);
         }
 
+        let mut container = container_builder.build();
+        container.env = env;
```

Comment:
This is prone to produce bad behavior. You can set env variables in the container builder. The way you do it here overwrites anything added before (i know not the case now, but could be in the future).

Reply:
I can't set env variables loaded from secrets I believe, or can I?

Reply:
Okay actually I think I might have found something